cast/src/lib.rs

```diff
@@ -628,6 +628,183 @@ where
         Ok(receipt)
     }
 
+    pub async fn events(&self, hash: String, chain: Chain, api_key: String) -> eyre::Result<String> {
+
+        // Get the receipt from the tx hash and pull the logs.
+        let receipt = self.receipt(hash, None, 0, false, true).await;
+        let receipt: serde_json::Value = match receipt {
```

Comment:
We also have a type for transaction receipts: https://docs.rs/ethers/latest/ethers/types/struct.TransactionReceipt.html